QgisModelBaker/gui/workflow_wizard/workflow_wizard.py

```diff
@@ -505,12 +505,15 @@ def get_topping_file_model(self, id_list):
             for downloaded_file_id in topping_file_cache.downloaded_files:
                 if downloaded_file_id in missing_file_ids:
                     missing_file_ids.remove(downloaded_file_id)
-            self.log_panel.print_info(
-                self.tr(
-                    "- - Some topping files where not successfully downloaded: {}"
-                ).format(" ".join(missing_file_ids)),
-                LogColor.COLOR_TOPPING,
-            )
+            try:
+                self.log_panel.print_info(
+                    self.tr(
+                        "- - Some topping files where not successfully downloaded: {}"
+                    ).format(" ".join(missing_file_ids)),
+                    LogColor.COLOR_TOPPING,
+                )
+            except Exception:
+                pass
```

Comment:
instead of `pass` could be a QgsMessageLog with the same text that would have been shown in the log panel?
Or what error could have been raised here?

Reply:
Uh oh. You found my dirty hack.

When downloading a file we do that in another thread. If the user closes the dialog this thread is still running and on a failure of the download the dialog does not exist anymore. This is similar to the situation with the running java thread (maybe you remember). This would be a bigger effort to fix. Here we just pass. Because when the user closes the dialog, she/he is not interested in the feedback. But yes, a QgsMessageLog would be appropriate.

Reply:
Hmm.. But this does not work. It's the whole object that is deleted on closing the dialog. So no option to write the LogMessage

Reply:
I leave it like this. There is an issue open to solve all this threading problem. Thing is, that the user encounters it very rarely. I encounter it more often while testing.